### PR TITLE
memory-profiler: Convert to V8 Heap Dump

### PR DESCRIPTION
**Pull request by keitwb**
This replaces the old output format in favor of a V8 Heap Snapshot
which is usable in Chrome Dev Tools.

This will hopefully be useful in debugging heap usage even though
it is not quite as full of metadata compared to the heap snapshots taken
from a normal V8 app.

This also adds the ability to export the heap snapshot to GEXF for use
in graphing tools like Gephi.

Also included is the ability to launch a REPL to manually explore the heap
graph.

Signed-off-by: Ben Keith <keitwb@gmail.com>

### CODE REVIEW

File: packages/memory-profiler/README.md [hunk 1]
```diff
@@ -3,5 +3,46 @@
...
+First, make a heap snapshot with the hidden sdk-cli command `heap-snapshot`.  Then,
+
+```
+$ node lib/cli.js v8 /path/to/fitbit-project/js-heap.bin /path/to/fitbit-project/build/app.fba /path/to/fitbit.heapsnapshot atlas
+```
+
+(replacing the paths and `atlas` with your device type).
+
+Then load the file `fitbit.heapsnapshot` in Chrome Dev Tools under the Memory tab. You can also take
+successive snapshots and load them in CHrome Dev Tools and compare them to see how the heap has
```
Comment (Hexxeh) [inline packages/memory-profiler/README.md:34]: typo nit: Chrome

**Review by Hexxeh — APPROVED** [review]
I'll look into the CI error since I don't think it's related and get this merged, thank you! :)

File: packages/memory-profiler/README.md (hunk 1 above)
Comment (keitwb) [reply packages/memory-profiler/README.md]: Fixed.

**Comment by Hexxeh** [comment]
This should be good for a rebase, I've rebased and run `yarn checkstyle -w` to run the formatter on another branch and it all looks happy. I'd do it, but I don't think I have access to change the branch this PR is from. If you'd prefer I can do it for you in another PR and keep you in the commit author/signed-off?

**Comment by keitwb** [comment]
> This should be good for a rebase, I've rebased and run `yarn checkstyle -w` to run the formatter on another branch and it all looks happy. I'd do it, but I don't think I have access to change the branch this PR is from. If you'd prefer I can do it for you in another PR and keep you in the commit author/signed-off?

Ok, I rebased.

**Comment by Hexxeh** [comment]
> > This should be good for a rebase, I've rebased and run `yarn checkstyle -w` to run the formatter on another branch and it all looks happy. I'd do it, but I don't think I have access to change the branch this PR is from. If you'd prefer I can do it for you in another PR and keep you in the commit author/signed-off?
> 
> Ok, I rebased.

You'll need to run `yarn checkstyle -w` as well to pass CI, there's normally a Git hook that does this automatically but I'm not sure why it hasn't run it in your case. Sorry about that!

**Comment by keitwb** [comment]
ok, yeah I think the hooks didn't run because I only did a rebase instead of a regular commit.  Should be good now.

**Pull request merged**